src/imputation/MoR.py

```diff
@@ -112,8 +121,108 @@ def carry_forwards(df, backdata, impute_vars):
         df.loc[match_cond, f"{var}_imputed"] = df.loc[match_cond, f"{var}_prev"]
     df.loc[match_cond, "imp_marker"] = "CF"
 
+    df.loc[match_cond] = create_imp_class_col(df,
+                                              "200_prev",
+                                              "201_prev",
+                                              use_cellno=False)
+
     # Drop merge related columns
-    to_drop = [column for column in df.columns if column.endswith("_prev")]
+    to_drop = [column for column in df.columns
+               if (column.endswith("_prev")) & (column not in impute_vars)]
     to_drop += ["_merge"]
     df = df.drop(to_drop, axis=1)
     return df
+
+
+def calculate_links(current_df, prev_df, target_vars):
+    """Calculate the links between previous and current data.
+
+    Args:
+        current_df (_type_): _description_
+        prev_df (_type_): _description_
+        target_vars (_type_): _description_
+    """
+    link_df = pd.merge(current_df,
+                       prev_df,
+                       on=["reference", "imp_class"],
+                       how="inner",
+                       suffixes=("", "_prev"))
+    target_vars_prev = [f"{var}_prev" for var in target_vars]
+    link_df = (link_df[["reference", "imp_class"] + target_vars + target_vars_prev]
+               .groupby(["reference", "imp_class"])
+               .sum()
+               )
+    # Calculate the ratios for the relevant variables
+    for target in target_vars:
+        link_df[f"{target}_link"] = link_df[target] / link_df[f"{target}_prev"]
+
+    # TODO How to deal with infs or 0/0
```

Comment:
suggestion: could use np.where like in imputation.apportionment.py, or .loc to avoid division by zero.

Reply:
I think this definitely a good shout for clarity, but I don't think we need to do it as we get rid of the NaNs produced by zero division using the `no_zero_mask ` in line 222.